File: AkkaWithZeroMQ/src/main/scala/org/akka/essentials/zeromq/example1/PublisherActor.scala
package org.akka.essentials.zeromq.example1
import akka.actor.Actor
import akka.actor.ActorLogging
import akka.util.duration._
import akka.zeromq.Bind
import akka.zeromq.SocketType
import akka.zeromq.ZeroMQExtension
import akka.zeromq._
import akka.actor.Cancellable

case class Tick

class PublisherActor extends Actor with ActorLogging {
  val pubSocket = ZeroMQExtension(context.system).newSocket(SocketType.Pub, Bind("tcp://127.0.0.1:1234"))
  var count = 0
  var cancellable:Cancellable = null
  override def preStart() {
    cancellable = context.system.scheduler.schedule(1 second, 1 second, self, Tick)
  }
  def receive: Receive = {
    case Tick =>
      count += 1
      var payload = "This is the workload " + count;
      pubSocket ! ZMQMessage(Seq(Frame("someTopic"), Frame(payload)))
      if(count == 10){
      	cancellable.cancel()
      }
  }
}

File: AkkaWithZeroMQ/src/main/java/org/akka/essentials/zeromq/example1/PublisherActor.java
package org.akka.essentials.zeromq.example1;

import akka.actor.ActorRef;
import akka.actor.Cancellable;
import akka.actor.UntypedActor;
import akka.util.Duration;
import akka.zeromq.Bind;
import akka.zeromq.Frame;
import akka.zeromq.ZMQMessage;
import akka.zeromq.ZeroMQExtension;

public class PublisherActor extends UntypedActor {
	public static final Object TICK = "TICK";
	int count = 0;
	Cancellable cancellable;
	ActorRef pubSocket = ZeroMQExtension.get(getContext().system())
			.newPubSocket(new Bind("tcp://127.0.0.1:1237"));

	@Override
	public void preStart() {
		cancellable = getContext()
				.system()
				.scheduler()
				.schedule(Duration.parse("1 second"),
						Duration.parse("1 second"), getSelf(), TICK);
	}

	@Override
	public void onReceive(Object message) throws Exception {
		if (message.equals(TICK)) {
			pubSocket.tell(new ZMQMessage(new Frame("someTopic"), new Frame(
					"This is the workload " + ++count)));
			
			if(count==10) 
				cancellable.cancel();
		}
	}
}
